src/hasi/segmentation_pipeline.py

```diff
@@ -0,0 +1,350 @@
+        print('landmarks_list:', landmarks_list)
+        sys.exit(1)
+    print(f'List of cases to process: {data_list}')
+
+    atlas_image_filename = root_dir + bone + '/' + atlas + '-AA.nrrd'
+    print(f'Reading atlas image from file: {atlas_image_filename}')
+    atlas_aa_image = itk.imread(atlas_image_filename)
+    # atlas_aa_segmentation = itk.imread(root_dir + bone + '/' + atlas + '-AA.seg.nrrd',
+    #                                    pixel_type=itk.VariableLengthVector[itk.UC])
+    atlas_labels_filename = root_dir + bone + '/' + atlas + '-AA-label.nrrd'
+    print(f'Reading atlas labels from file: {atlas_labels_filename}')
+    atlas_aa_segmentation = itk.imread(atlas_labels_filename, pixel_type=itk.UC)
+
+    print('Computing bounding box of the labels')
+    # reduce the image to a bounding box around the segmented bone
+    # the other content makes the registration more difficult
+    # because the knees will be bent to different degree etc
+    atlas_bounding_box = label_bounding_box(atlas_aa_segmentation)
+
+    atlas_aa_segmentation = itk.region_of_interest_image_filter(
+        atlas_aa_segmentation,
+        region_of_interest=atlas_bounding_box)
+    atlas_bone_label_filename = root_dir + bone + '/' + atlas + '-AA-' + bone + '-label.nrrd'
+    print(f'Writing {bone} variant of atlas labels to file: {atlas_bone_label_filename}')
+    itk.imwrite(atlas_aa_segmentation, atlas_bone_label_filename, compression=True)
+
+    atlas_aa_image = itk.region_of_interest_image_filter(
+        atlas_aa_image,
+        region_of_interest=atlas_bounding_box)
+    atlas_bone_image_filename = root_dir + bone + '/' + atlas + '-AA-' + bone + '.nrrd'
+    print(f'Writing {bone} variant of atlas image to file: {atlas_bone_image_filename}')
+    itk.imwrite(atlas_aa_image.astype(itk.SS), atlas_bone_image_filename)
+
+    csv_filename = root_dir + bone + '/' + atlas + '-BoneMorphometry.csv'
+    with open(csv_filename, 'w') as morphometry_csv:
+        morphometry_csv.write('Bone,Atlas,Case,Label,BVTV,TbN,TbTh,TbSp,BSBV\n')
+
+    # now go through all the cases, doing main processing
+    for case in data_list:
+        print(u'\u2500' * 80)
+        print(f'Processing case {case}')
+
+        # process_case(root_dir, bone, case, bone_label, atlas)
+
+        # Elastix crashes on second iteration of this for loop,
+        # so we use subprocess as a work-around
+        status = subprocess.run(['python', __file__, root_dir, bone, case, str(bone_label), atlas])
+        if status.returncode != 0:
+            print(f'Case {case} failed with error {status.returncode}')
+        else:
+            print(f'Success processing case {case}')
+
+    print(f'Processed {len(data_list)} cases for bone {bone} using {atlas} as atlas.\n\n\n')
+
+
+if __name__ == '__main__':
+    if len(sys.argv) == 1:  # direct invocation
+        atlas_list = ['901-L', '901-R', '907-L', '907-R', '917-L', '917-R', 'F9-3wk-02-L', 'F9-3wk-02-R']
```

Comment:
Perhaps not necessary to change for this PR, but would be good to clarify what steps will be needed to generalize pipeline for arbitrary samples. We probably would not want a developer to have to import these lines even if they never run `__main__` directly.

Reply:
Currently, this is the entry-point for the analysis. We plan to use https://github.com/dagster-io/dagster later, so this will have to change too.